api/server.go

```diff
@@ -263,6 +264,43 @@ func (s *Server) DownloadVault(c echo.Context) error {
 	return c.File(filePathName)
 }
 
+func (s *Server) GetVault(c echo.Context) error {
+	publicKeyECDSA := c.Param("publicKeyECDSA")
+	if publicKeyECDSA == "" {
+		return fmt.Errorf("public key is required")
+	}
+
+	filePathName := filepath.Join(s.vaultFilePath, publicKeyECDSA+".bak")
+	_, err := os.Stat(filePathName)
+	if err != nil {
+		return fmt.Errorf("fail to get file info, err: %w", err)
+	}
+
+	passwd := c.Request().Header.Get("x-password")
+	if passwd == "" {
+		return fmt.Errorf("vault backup password is required")
+	}
+
+	content, err := os.ReadFile(filePathName)
+	if err != nil {
+		return fmt.Errorf("fail to read file, err: %w", err)
+	}
+
+	vault, err := common.DecryptVaultFromBackup(passwd, content)
+	if err != nil {
+		return fmt.Errorf("fail to decrypt vault from the backup, err: %w", err)
+	}
+
+	return c.JSON(http.StatusOK, types.VaultGetResponse{
+		Name:           vault.Name,
+		PublicKeyEcdsa: vault.PublicKeyEcdsa,
+		PublicKeyEddsa: vault.PublicKeyEddsa,
+		HexChainCode:   vault.HexChainCode,
+		KeyShares:      vault.KeyShares,
```

Comment:
can you please not return "keyshares" , because client side doesn't need keyshares

Reply:
done